src/lib/y2security/lsm/config.rb

```diff
@@ -0,0 +1,152 @@
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+require "y2security/lsm/base"
+require "y2security/lsm/app_armor"
+require "y2security/lsm/none"
+require "y2security/lsm/selinux"
+
+Yast.import "Stage"
+Yast.import "ProductFeatures"
+
+module Y2Security
+  module LSM
+    # This class allows to check and select one of the supported Linux Security Modules (LSM)
+    class Config
+      include Yast::Logger
+      include Singleton
+
+      RUNNING_PATH = "/sys/kernel/security/lsm".freeze
+      SUPPORTED = [None, Selinux, AppArmor].freeze
+
+      # @return [AppArmor,Selinux, nil] selected module
+      attr_accessor :selected
+      # @returb [Boolean] Whether LSM can be configured by the user or not
+      attr_accessor :configurable
+
+      def initialize
+        supported.each { |m| self.class.send(:define_method, m.id) { m } }
+      end
+
+      # Select the LSM to be used based in the one defined in the control file using apparmor as
+      # fallback in case that no one is selected
+      def propose_default
+        log.info("The settings are #{product_feature_settings.inspect}")
+        selected = product_feature_settings.fetch(:default, "apparmor")
+
+        select(selected)
+      end
+
+      # Select the module with the given id when it is selectable.
+      #
+      # @param id [String,Symbol] LSM id to be selected
+      def select(id)
+        @selected = selectable.find { |m| m.id == id.to_sym }
+        log.info @selected ? "Selected LSM with id: #{id}" : "No LSM selected with id: #{id}"
+        @selected
+      end
+
+      # Convenience method to obtain a list of the supported and selectable modules
+      #
+      # @return [Array<Y2Security::LSM::Base] array of supported and selectable LSMs
+      def selectable
+        supported.select(&:selectable?)
+      end
+
+      # Returns the needed patterns for the selected LSM or an empty array if no one is selected
+      #
+      # @return [Array<Sting>]
+      def needed_patterns
+        return [] unless selected
+
+        selected.needed_patterns
+      end
+
+      # Convenience method to save the configuration for the selected LSM
+      def save
+        return false unless selected
+
+        selected.save
+      end
+
+      # Obtains the supported and active Linux Security Major Module from the running system
+      #
+      # @return [Y2Security::LSM::Base]
+      def from_system
+        active.first
+      end
+
+      # Return an array with the supported and active Linux Security Major Modules
+      #
+      # @return [Array<Y2Security::LSM::Base>]
+      def active
+        return [] unless Yast::Stage.normal
+
+        modules = Yast::SCR.Read(Yast.path(".target.string"), RUNNING_PATH)
```

Comment:
I prefer to use [`Yast::Execute`](https://github.com/yast/yast-yast2/blob/master/library/system/src/lib/yast2/execute.rb)

Reply:
Originally I used Yast::Execute but then I doubt and moved back to the `Yast::SCR.Read`